packages/i18next/src/i18nextPlugin.ts

```diff
@@ -0,0 +1,116 @@
+import { Plugin, isNil, type Runtime } from "@squide/core";
+import type { i18n } from "i18next";
+import LanguageDetector, { type DetectorOptions } from "i18next-browser-languagedetector";
+import { i18nextInstanceRegistry } from "./i18nextInstanceRegistry.ts";
+
+export interface i18nextPluginOptions {
+    detection?: Omit<DetectorOptions, "lookupQuerystring">;
+}
+
+export function findSupportedLanguage<T>(userLanguages: string[], supportedLanguages: T[]) {
```

Comment:
I was finding it weird that findSupportedLanguage took supportedLanguages as an input. What's the goal here, finding the optimal language for a user?

`userLanguages` should also be renamed to `userPreferredLanguages`

Reply:
It's tied to the user preferred languages as it's also used by the `changeLanguage` method. Basically, it's a function that returns the first language in `userLanguages` that is included in `supportedLanguages`.

Reply:
Renamed `userLanguages` for `userPreferredLanguages` and also renamed the function to `findSupportedPreferredLanguage`.